File: 643-maximum-average-subarray-i/maximum-average-subarray-i.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        ans=-(1<<31)
        preSum=0
        for i in range(k):
            preSum=preSum+nums[i]
        ans=max(ans,preSum/k)
        for i in range(k,len(nums)):
            preSum=preSum-nums[i-k]
            preSum=preSum+nums[i]
            ans=max(ans,preSum/k)
        return ans

        

File: 643-maximum-average-subarray-i/maximum-average-subarray-i.java
class Solution {
    public double findMaxAverage(int[] nums, int k) {
        if(nums.length<k){
            return 0.0;
        }
        double maxi=-1000000;
        int sum=0;
        for(int i=0;i<k;i++){
            sum=sum+nums[i];
        }
        maxi=Math.max(maxi,(double)sum/k);
        for(int i=k;i<nums.length;i++){
            sum-=nums[i-k];
            sum+=nums[i];
            maxi=Math.max(maxi,(double)sum/k);
        }
        return maxi;
    }
}